satpy/readers/viirs_sdr.py

```diff
@@ -248,25 +248,43 @@ def scale_swath_data(self, data, scaling_factors):
         data = data * factors[:, 0] + factors[:, 1]
         return data
 
-    def adjust_scaling_factors(self, factors, file_units, output_units):
-        """Adjust scaling factors."""
-        if file_units == output_units:
-            LOG.debug("File units and output units are the same (%s)", file_units)
-            return factors
-        if factors is None:
-            return None
-        factors = factors.where(factors != -999., np.float32(np.nan))
-
+    @staticmethod
+    def _scale_factors_for_units(factors, file_units, output_units):
         if file_units == "W cm-2 sr-1" and output_units == "W m-2 sr-1":
-            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'", file_units, output_units)
+            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'",
+                      file_units, output_units)
             factors = factors * 10000.
-            return factors
         elif file_units == "1" and output_units == "%":
-            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'", file_units, output_units)
+            LOG.debug("Adjusting scaling factors to convert '%s' to '%s'",
+                      file_units, output_units)
             factors = factors * 100.
-            return factors
         else:
+            LOG.debug("Not sure how to perform unit conversion '%s' to '%s'",
+                      file_units, output_units)
+        return factors
+
+    @staticmethod
+    def _get_valid_scaling_factors(factors):
+        if factors is None:
+            factors = np.array([1, 0], dtype=np.float32)
+            factors = xr.DataArray(da.from_array(factors, chunks=1))
+        else:
+            factors = factors.where(factors != -999., np.float32(np.nan))
+        return factors
+
+    def _adjust_scaling_factors(self, factors, file_units, output_units):
+        """Adjust scaling factors ."""
+        if file_units == output_units:
+            LOG.debug("File units and output units are the same (%s)", file_units)
             return factors
+        factors = self._get_valid_scaling_factors(factors)
+        return self._scale_factors_for_units(factors, file_units, output_units)
+
+    def _get_scaling_factors(self, file_units, output_units, factor_var_path):
+        """Get file scaling factors and scale according to expected units."""
+        factors = self.get(factor_var_path)
+        factors = self._adjust_scaling_factors(factors, file_units, output_units)
+        return factors
```

Comment:
Nice refactoring! It is fine as it is, but I have a couple of suggestion that can make this even better I think:
In `_get_scaling_factors`, getting the factors from the file isn't really at the same level of abstraction as calling the adjusting function. So how about letting `_get_valid_scaling_factors` do that and use it in `_get_scaling_factors`, eg:
```python
    def _get_valid_scaling_factors(self, factor_var_path):
        factors = self.get(factor_var_path)
        if factors is None:
            factors = np.array([1, 0], dtype=np.float32)
            factors = xr.DataArray(da.from_array(factors, chunks=1))
        else:
            factors = factors.where(factors != -999., np.float32(np.nan))
        return factors

    def _adjust_scaling_factors(self, factors, file_units, output_units):
        """Adjust scaling factors ."""
        if file_units == output_units:
            LOG.debug("File units and output units are the same (%s)", file_units)
            return factors
        return self._scale_factors_for_units(factors, file_units, output_units)

    def _get_scaling_factors(self, file_units, output_units, factor_var_path):
        """Get file scaling factors and scale according to expected units."""
        factors = self._get_valid_scaling_factors(factor_var_path)
        factors = self._adjust_scaling_factors(factors, file_units, output_units)
        return factors
```

Reply:
Note however that with this change, `factors` will never be None in `_adjust_scaling_factors` anymore, but it looks cleaner this way.

Reply:
Good point. I avoided it because of the `factors` not being `None` any more and what that does for extra tasks on the dask graph. I'll think about it. I'm still trying to figure out the radiance units thing discussed on slack.